Document BuildWithNetFrameworkHostedCompiler and friends (#45620)
dotnet/docs: docs/core/project-sdk/msbuild-props.md

</PropertyGroup>
```

### BuildWithNetFrameworkHostedCompiler

When using .NET Framework MSBuild, `BuildWithNetFrameworkHostedCompiler=true` ensures that
a C#/VB compiler corresponding to the current SDK version is used
instead of the default version that ships with MSBuild.

> This uses the .NET Framework version of the compiler unlike `RoslynUseSdkCompiler`.

Reviewer: ```suggestion
When this property is set to `true`, the .NET Framework version of the compiler is used, unlike `RoslynUseSdkCompiler`.
```

> In some cases, this happens automatically when it is detected that MSBuild and SDK versions are different,

Reviewer: ```suggestion
In some cases, this behavior happens automatically when it is detected that MSBuild and SDK versions are different,
```

> and then `BuildWithNetFrameworkHostedCompiler=false` can be used to opt out of this behavior.

Reviewer: ```suggestion
and then you can set `BuildWithNetFrameworkHostedCompiler=false` to opt out of the behavior.
```

### RoslynUseSdkCompiler

When using .NET Framework MSBuild, `RoslynUseSdkCompiler=true` ensures that
a C#/VB compiler corresponding to the current SDK version is used
instead of the default version that ships with MSBuild.

> This uses the .NET Core version of the compiler unlike `BuildWithNetFrameworkHostedCompiler`.

Reviewer: ```suggestion
When this property is set to `true`, the .NET Core version of the compiler is used, unlike `BuildWithNetFrameworkHostedCompiler`.
```

In most cases, `RoslynUseSdkCompiler=true` is the default setting.

### RoslynUseMSBuildCompiler

> `RoslynUseMSBuildCompiler=true` can be used to opt out of an implicit `RoslynUseSdkCompiler=true`.

Reviewer 2: Interesting. Why is this property necessary? Isn't setting `<RoslynUseSdkCompiler>false</RoslynUseSdkCompiler>` sufficient?

Author: Yes, it should be sufficient. RoslynUseMSBuildCompiler was documented by @jaredpar in https://github.com/dotnet/sdk/blob/main/documentation/general/decouple-vs-and-net-sdk.md that's why I included it in the implementation and here, but I agree it feels like we might not need it.
